pkg/cache/json.go

```diff
@@ -58,31 +60,66 @@ func (q *JSON) ListBundles(ctx context.Context) ([]*api.Bundle, error) {
 }
 
 func (q *JSON) SendBundles(_ context.Context, s registry.BundleSender) error {
+	var keys []apiBundleKey
 	for _, pkg := range q.packageIndex {
-		channels := sets.KeySet(pkg.Channels)
-		for _, chName := range sets.List(channels) {
-			ch := pkg.Channels[chName]
-
-			bundles := sets.KeySet(ch.Bundles)
-			for _, bName := range sets.List(bundles) {
-				b := ch.Bundles[bName]
-				apiBundle, err := q.loadAPIBundle(apiBundleKey{pkg.Name, ch.Name, b.Name})
-				if err != nil {
-					return fmt.Errorf("convert bundle %q: %v", b.Name, err)
-				}
-				if apiBundle.BundlePath != "" {
-					// The SQLite-based server
-					// configures its querier to
-					// omit these fields when
-					// bundle path is set.
-					apiBundle.CsvJson = ""
-					apiBundle.Object = nil
-				}
-				if err := s.Send(apiBundle); err != nil {
-					return err
-				}
+		for _, ch := range pkg.Channels {
+			for _, b := range ch.Bundles {
+				keys = append(keys, apiBundleKey{pkg.Name, ch.Name, b.Name})
+			}
+		}
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		if keys[i].chName != keys[j].chName {
+			return keys[i].chName < keys[j].chName
+		}
+		if keys[i].pkgName != keys[j].pkgName {
+			return keys[i].pkgName < keys[j].pkgName
+		}
+		return keys[i].name < keys[j].name
+	})
+	var files []*os.File
+	var readers []io.Reader
+	for _, key := range keys {
+		filename, ok := q.apiBundles[key]
+		if !ok {
+			return fmt.Errorf("package %q, channel %q, key %q not found", key.pkgName, key.chName, key.name)
+		}
+		file, err := os.Open(filename)
```

Comment:
This means we'd have open files for every bundle in the catalog at once, so in the 1000s roughly. Is that reasonable? If that's undesirable, we could in theory have an `io.Pipe` with the pipe reader passed into the decoder. And then a separate goroutine opens and writes a file at a time into the pipe.

Reply:
Sgtm. We can revisit later if it turns out not to be fine.